### PR TITLE
adds proper gibtonite detonation timing

### LINKED ISSUE #28071: Non-mineral gibtonite is instantlyexploding
WHAT THE FUCK DID YOU DO

Shoot the rock with your ka stop the detonation at 0 seconds.

Move rock and hit with your pickaxe and prepare to run but wait IT INSTANTLY FUCKING EXPLODES of having 10 second countdown

### PR DESCRIPTION
**Pull request by silicons**
nice one @optimumtact 
fixes #28071

### CODE REVIEW

File: code/modules/mining/ores_coins.dm [hunk 1]
```diff
@@ -188,6 +188,7 @@
 	var/det_time = 100
```
Comment (Cyberboss) [inline code/modules/mining/ores_coins.dm]: One line above this, set primed to FALSE please

File: code/modules/mining/ores_coins.dm [hunk 2]
```diff
@@ -260,14 +263,17 @@
...
 			log_game("[key_name(user)] has primed a [name] for detonation at [A][COORD(bombturf)]")
-		if(primed)
-			if(quality == 3)
-				explosion(src.loc,2,4,9,adminlog = notify_admins)
-			if(quality == 2)
-				explosion(src.loc,1,2,5,adminlog = notify_admins)
-			if(quality == 1)
-				explosion(src.loc,-1,1,3,adminlog = notify_admins)
-			qdel(src)
+		det_timer = addtimer(CALLBACK(src, .proc/detonate, notify_admins), det_time)
```
Comment (Cyberboss) [inline code/modules/mining/ores_coins.dm]: Needs TIMER_STOPPABLE flag

File: code/modules/mining/ores_coins.dm [hunk 3]
```diff
@@ -260,14 +263,17 @@
...
+	
+/obj/item/weapon/twohanded/required/gibtonite/proc/detonate(notify_admins)
+	if(primed)
+		if(quality == 3)
```
Comment (Cyberboss) [inline code/modules/mining/ores_coins.dm]: use a switch and make these defines

File: code/modules/mining/ores_coins.dm [hunk 4]
```diff
@@ -260,14 +263,17 @@
...
+			explosion(src.loc,2,4,9,adminlog = notify_admins)
```
Comment (Cyberboss) [inline code/modules/mining/ores_coins.dm]: explosion(src,...

File: code/modules/mining/ores_coins.dm [hunk 5]
```diff
@@ -260,14 +263,17 @@
...
+		else if(quality == 2)
+			explosion(src.loc,1,2,5,adminlog = notify_admins)
+		else if(quality == 1)
+			explosion(src.loc,-1,1,3,adminlog = notify_admins)
```
Comment (Cyberboss) [inline code/modules/mining/ores_coins.dm]: -1 devastation does nothing, set it to 0

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/modules/mining/ores_coins.dm (hunk 3 above)
Comment (lzimann) [inline code/modules/mining/ores_coins.dm]: Pretty sure gibtonite has some defines already.

**Comment by Kromgar** [comment]
Good god please we need this

File: code/modules/mining/ores_coins.dm (hunk 1 above)
Comment (kevinz000) [inline code/modules/mining/ores_coins.dm]: done

**Pull request merged**